### PR TITLE
JS2 week3 hw

### PR DESCRIPTION
**Pull request by amudar**
Hi,
Please find my homework for your review.
Cheers - Mudar

### CODE REVIEW

File: Week3/homework/2-step3.js [hunk 1]
```diff
@@ -2,7 +2,34 @@
...
+    }
+
+    if (i % 3 === 0 && i % 5 === 0) {
+      console.log(i);
+    }
+
+  }
+
+  return values;
+
```
Comment (rohni) [inline Week3/homework/2-step3.js]: Nicely done, and laid out.  Easy to read.  And you only make one run through the array.

File: Week3/homework/3-step3.js [hunk 2]
```diff
@@ -2,24 +2,47 @@
...
+
+  if (num > 0) {
+    do {
+      myStr += str;
+      num--;
+    } while (num > 0);
+  }
+
+  return myStr;
 }
```
Comment (rohni) [inline Week3/homework/3-step3.js]: 👍 Great job.  Something I noticed in some of the other student's work was that they also had a check to make sure num > 0.
I also did not do this, but I thought it was a good idea.

File: Week3/homework/4-step3.js [hunk 3]
```diff
@@ -1,2 +1,13 @@
...
+// Here is the link: https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/object-oriented-programming/use-a-constructor-to-create-objects
+
+function Dog() {
+    this.name = "Rupert";
+    this.color = "brown";
+    this.numLegs = 4;
+}
+// Add your code below this line
+const hound = new Dog();
+console.log(hound);
```
Comment (rohni) [inline Week3/homework/4-step3.js]: ✅

File: Week3/homework/5-step3.js [hunk 4]
```diff
@@ -1,2 +1,18 @@
...
+    }
+    // Only change code above this line
+    console.log(product);
+    return product;
+}
+
+// Modify values below to test your code
+multiplyAll([[1, 2], [3, 4], [5, 6, 7]]);
+
+
```
Comment (rohni) [inline Week3/homework/5-step3.js]: ✅

File: Week3/homework/6-step3.js [hunk 5]
```diff
@@ -4,3 +4,12 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
...
 // add your solution here, or add a comment on how you would tackle this problem
+function print3D(myArr) {
+    if (Array.isArray(myArr)) {
+        myArr.forEach(elem => print3D(elem));
+    } else {
+        console.log(myArr);
+    }
+}
+
+print3D(arr3d);
```
Comment (rohni) [inline Week3/homework/6-step3.js]: ✅ Nice work, and this will work for any depth of array.

File: Week3/homework/7-step3.js [hunk 6]
```diff
@@ -21,5 +21,7 @@ f2(y);
 
 console.log(y);
 
-// Add your explanation as a comment here
+// In the first function we have "pass by value" where the variable value does not change,
+// while in the second it is "pass by reference" where the function is modifying the property value.
+
```
Comment (rohni) [inline Week3/homework/7-step3.js]: ✅

File: Week3/homework/step4-bonus.js [hunk 7]
```diff
@@ -5,6 +5,6 @@ const values = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
 // Add your function here. Try and come up with a good name for this function
 
 // Replace `yourFunction` with the name of the function you just created
-const uniqueValues = yourFunction(values);
+const uniqueValues = [...(new Set(values))];
 
 console.log(uniqueValues);
```
Comment (rohni) [inline Week3/homework/step4-bonus.js]: 👍 Nicely done.  By far the easiest solution and the fastest. 😄

File: Week3/homework/step4.js [hunk 8]
```diff
@@ -1,8 +1,15 @@
...
 
 const addSix = createBase(6);
 
-addSix(10); // returns 16
-addSix(21); // returns 27
+addSix(10);
+addSix(21);
+
+console.log(addSix(10)); // returns 16
+console.log(addSix(21)); // returns 27
```
Comment (rohni) [inline Week3/homework/step4.js]: ✅

**Comment by rohni** [comment]
Good work amudar! 😄 
Nice solutions, and very readable.

**Comment by amudar** [comment]
Rohni,
Thank you very much for your time to review and feedback.
Cheers-Mudar

**Pull request closed without merging**